### PR TITLE
Improve handling of DateTime property bag properties

### PR DESCRIPTION
**Pull request by lgolding**
Fix a bug in `PropertyBagConverter` that prevented `DateTime`-valued property bag properties from round-tripping correctly. With this fix in place, it is no longer necessary to work around the problem by circumventing Newtonsoft.Json's handling of `DateTime`s, and so the helper method `SarifUtilities.DeserializeObject` is no longer needed.

Revert calls to the deleted method to their state before the method was introduced. In most cases, the previous call was to `JsonConvert.DeserializeObject`. In one case, the previous call was to `JObject.Parse`.

### CODE REVIEW

File: src/Test.UnitTests.Sarif/TestData/RoundTripping/RoundTripping.sarif [hunk 1]
```diff
@@ -26,7 +26,7 @@
             "dateTimeArray": [
   "2019-09-26T15:52:02",
   "2019-09-26T15:54",
-  "2019-09-26T15:55:05.123456789"
+  "2019-09-26T15:55:05.1234567"
```
Comment (lgolding) [inline src/Test.UnitTests.Sarif/TestData/RoundTripping/RoundTripping.sarif]: >1234567 [](start = 23, length = 7)

Now that we once again rely on Newtonsoft.Json's serialization and deserialization of `DateTime`s, they only round-trip 7 fractional second digits.

**Comment by lgolding** [comment]
@michaelcfanning is added to the review. #Closed

**Comment by lgolding** [comment]
@ScottLouvau is added to the review. #Closed

File: src/Sarif/Readers/PropertyBagConverter.cs [hunk 2]
```diff
@@ -49,6 +49,12 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                     serializedValue = JsonConvert.ToString(serializedValue);
                     isString = true;
                 }
+                else if (propertyType == typeof(DateTime))
```
Comment (lgolding) [inline src/Sarif/Readers/PropertyBagConverter.cs]: >else [](start = 16, length = 4)

The fix. A simple `else` might also work but this is the minimal change that fixes the bug.

File: src/Test.UnitTests.Sarif/TestData/RoundTripping/RoundTripping.sarif (hunk 1 above)
Comment (ScottLouvau) [reply src/Test.UnitTests.Sarif/TestData/RoundTripping/RoundTripping.sarif]: That seems sufficient.

**Pull request merged**